transaction-view/src/resolved_transaction_view.rs

```diff
@@ -0,0 +1,482 @@
+                    loaded_address_index < num_writable_lookup_accounts
+                } else if index >= num_signed_accounts {
+                    let unsigned_account_index = index.wrapping_sub(num_signed_accounts);
+                    unsigned_account_index < num_writable_unsigned_accounts
+                } else {
+                    index < num_writable_signed_accounts
+                }
+            };
+
+            // If the key is reserved it cannot be writable.
+            is_writable_cache.push(is_requested_write && !reserved_account_keys.contains(key));
+        }
+
+        // If the upgradable loader is not present and a key is called as a program
+        // it is not writable.
+        // If the upgradable loader is present, then there is no point to loop.
+        // Looping over the instructions is more efficient than looping over each account
+        // and checking `is_invoked` which loops over each instruction internally.
+        if !is_upgradable_loader_present {
+            for ix in view.instructions_iter() {
+                is_writable_cache[usize::from(ix.program_id_index)] = false;
+            }
+        }
+
+        is_writable_cache
+    }
+
+    fn num_readonly_accounts(&self) -> usize {
+        usize::from(
+            self.view
+                .total_readonly_lookup_accounts()
+                .wrapping_add(u16::from(self.view.num_readonly_signed_accounts()))
+                .wrapping_add(u16::from(self.view.num_readonly_unsigned_accounts())),
+        )
+    }
+
+    fn signature_details(&self) -> TransactionSignatureDetails {
+        // counting the number of pre-processor operations separately
+        let mut num_secp256k1_instruction_signatures: u64 = 0;
+        let mut num_ed25519_instruction_signatures: u64 = 0;
+        for (program_id, instruction) in self.program_instructions_iter() {
+            if secp256k1_program::check_id(program_id) {
+                if let Some(num_verifies) = instruction.data.first() {
+                    num_secp256k1_instruction_signatures =
+                        num_secp256k1_instruction_signatures.wrapping_add(u64::from(*num_verifies));
+                }
+            } else if ed25519_program::check_id(program_id) {
+                if let Some(num_verifies) = instruction.data.first() {
+                    num_ed25519_instruction_signatures =
+                        num_ed25519_instruction_signatures.wrapping_add(u64::from(*num_verifies));
+                }
+            }
+        }
+
+        TransactionSignatureDetails::new(
+            u64::from(self.view.num_required_signatures()),
+            num_secp256k1_instruction_signatures,
+            num_ed25519_instruction_signatures,
+        )
+    }
+}
+
+impl<D: TransactionData> SVMMessage for ResolvedTransactionView<D> {
+    fn num_total_signatures(&self) -> u64 {
+        self.signature_details().total_signatures()
+    }
```

Comment:
nit: I know this is the same name for the `SanitizedMessage` method but I can't help but suggest we rename this to something like `calculate_total_signatures` to be clear to the caller that there is a non trivial amount of calculation happening here. Or maybe cache the value?

Reply:
Fine with renaming to calculate - eventually the entire sig details will be cached with RuntimeTransaction

Reply:
🤔 actually when making the change this morning, I am thinking of not renaming.

Eventually this will be cached by `RuntimeTransaction`, but some downstream stuff such as fee calculation will just call this on an `SVMMessage` type, and whether its' cached or calculated is abstracted away.